### ISSUE TITLE
[youtube][watch]: Display playlist runtime in playlist panel of watch page

### ISSUE THREAD

**Issue by DarhkVoyd** [enrich-platforms/toppings#4]
## Issue Type
- [x] Feature Request

## Description
Currently, the playlist runtime information is displayed in the metadata section of the playlist page. However, it would be great to also show this information in the playlist panel of the watch video page for that playlist. This feature would provide easier access to playlist runtime details without the need to navigate to the playlist.

## Expected Behavior
When viewing a video within a playlist, the runtime information for that playlist should be displayed within the playlist panel on the watch page.

## Current Behavior
At present, the playlist runtime information is only visible in the metadata section of the playlist page. Users must navigate to this playlist page to view playlist runtime details.

**Comment by yakatyansh** [comment]
I would like to work on it

**Comment by DarhkVoyd** [comment]
Hey @yakatyansh,

Thank you for showing interest in contributing to Toppings. However, I want to inform you that this particular issue is currently on hold as we're waiting to address some design-related changes, and it's not part of our Hacktoberfest goals this year.

We appreciate your enthusiasm, and we encourage you to keep an eye on our project for future opportunities to contribute.

I am removing the "ready to take" label to reflect that.

**Comment by yakatyansh** [comment]
> We appreciate your enthusiasm, and we encourage you to keep an eye on our project for future opportunities to contribute.
> 
> I am removing the "ready to take" label to reflect that.

sure, will keep an eye on this issue and would love to create issues if I find any on testing

**Comment by Sandesh-03** [comment]
assign this issue to me

**Comment by DarhkVoyd** [comment]
@yakatyansh This issue is now ready to take, If you are still interested in working on it.

**Comment by yakatyansh** [comment]
Yes would love to

**Comment by DarhkVoyd** [comment]
@yakatyansh  I've assigned the issue to you. If you have any questions or need assistance along the way, feel free to ask.

**Comment by yakatyansh** [comment]
so sorry it is taking time, am sick lately, but working on the issue rn

**Comment by DarhkVoyd** [comment]
@yakatyansh Please don't worry about the timeline. Whenever you're feeling better and ready to continue, you can pick it up.

**Comment by yakatyansh** [comment]
Hey, I am better after a series of festivals and sickness, from today I will work on it properly, I just wanted to know that how to access the metadata section

**Comment by DarhkVoyd** [comment]
Hey, you could possibly start by looking at the implementation of `addMetadataToppings` in `routeType/playlist` to better understand how it was implemented there and them similarly implement for `routeType/watch`.
https://github.com/enrich-platforms/toppings/blob/e6e9070788afe8a412f57c5aa8cebee1647e225c/src/scripts/youtube/routeType/playlist/addMetadataToppings.ts#L20
You could also use these css styles implemented [here](https://github.com/enrich-platforms/toppings/blob/e6e9070788afe8a412f57c5aa8cebee1647e225c/src/css/youtube.css#L40) or write your own to better blend into YouTube's native UI.

**Comment by yakatyansh** [comment]
Thank you for the great explaination, I am now quite clear about what to do here
<img width="1440" alt="Screenshot 2023-10-28 at 2 50 17 PM" src="https://github.com/enrich-platforms/toppings/assets/90173866/7be262c5-cba7-44f4-8b87-9561379b6a20">
so, this piece of UI
<img width="419" alt="Screenshot 2023-10-28 at 2 51 03 PM" src="https://github.com/enrich-platforms/toppings/assets/90173866/1562dbda-3329-4278-9c65-a12d5a2920df">
needs to go here,

**Comment by DarhkVoyd** [comment]
Yes, exactly. You should make sure that the styles blend seamlessly in YouTube's UI.

**Comment by yakatyansh** [comment]
<img width="626" alt="image" src="https://github.com/enrich-platforms/toppings/assets/90173866/8edf98a6-9ab8-4659-94f8-92cfc53c9860">

what do you think about this implementation ?, I am trying new things, because it is my first time implementing something on the YouTube UI

**Comment by yakatyansh** [comment]
yes regularily forking it, and working on the draft pr, actually somehow I lost my progress, so will have to do it again 🥲

**Comment by DarhkVoyd** [comment]
Completed. Playlist runtime is now shown in the watch page playlist panel when viewing a video as part of a playlist.

**How it works:**
1. When `onWatchPage` runs and the URL has a `?list=` query parameter, `injectPlaylistRuntimeInWatchPanel(playlistId)` is called
2. The function checks `getCachedPlaylist(playlistId)` — if the user has previously visited the playlist page (which caches the data), the runtime info is available immediately
3. `resolveTarget` finds the playlist panel header using 4 strategy selectors
4. A small inline badge is injected with total runtime (⏱) and average runtime
5. Badge is idempotent — re-navigating within a playlist removes and re-inserts it cleanly

**Design:**
- Uses YouTube CSS custom properties for text colour (compatible with dark/light themes)
- Minimal visual footprint — two stats separated by a faint dot, 11px font
- No badge shown if playlist data isn't cached (user hasn't visited the playlist page yet in the current cache window)
